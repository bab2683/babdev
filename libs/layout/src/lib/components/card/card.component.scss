@import "~@angular/material/theming";

:host {
  display: block;
  position: relative;

  &.with-hover {
    @include mat-elevation-transition;

    &:hover {
      @include mat-elevation(8);
    }
  }
}
