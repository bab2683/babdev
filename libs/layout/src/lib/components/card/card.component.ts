import { Component, HostBinding, Input } from '@angular/core';

@Component({
  selector: 'babdev-card',
  templateUrl: './card.component.html',
  styleUrls: ['./card.component.scss']
})
export class CardComponent {
  @HostBinding('class.with-hover')
  @Input()
  public withHover: boolean;
}
