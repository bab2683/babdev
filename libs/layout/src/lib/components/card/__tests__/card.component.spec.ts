import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CardComponent } from '../card.component';

describe('CardComponent', () => {
  let component: CardComponent;
  let fixture: ComponentFixture<CardComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [CardComponent],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(CardComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    component.withHover = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    component.withHover = false;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
