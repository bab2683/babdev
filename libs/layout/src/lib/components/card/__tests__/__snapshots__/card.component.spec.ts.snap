// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent should match snapshot 1`] = `
<babdev-card
  withHover={[Function Boolean]}
>
  <mat-card />
</babdev-card>
`;

exports[`CardComponent should match snapshot 2`] = `
<babdev-card
  withHover="false"
>
  <mat-card />
</babdev-card>
`;
